components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90

```diff
@@ -2060,173 +1946,233 @@ subroutine ma_precpevap30_convproc(                         &
                               ! this is done to allow better tracking of the
                               !    resuspension in the mass-budget post-processing scripts
 
-   real(r8), intent(in)    :: rprd(pcols,pver)  ! conv precip production  rate (gathered)
-   real(r8), intent(in)    :: evapc(pcols,pver)  ! conv precip evaporation rate (gathered)
-   real(r8), intent(in)    :: dp_i(pver) ! pressure thickness of level (in mb)
+   real(r8), intent(in)    :: rprd(pcols,pver)  ! conv precip production  rate (gathered) [kg/kg/s]
+   real(r8), intent(in)    :: evapc(pcols,pver)  ! conv precip evaporation rate (gathered) [kg/kg/s]
+   real(r8), intent(in)    :: dp_i(pver) ! pressure thickness of level [mb]
 
    integer,  intent(in)    :: icol  ! normal (ungathered) i index for current column
    integer,  intent(in)    :: ktop  ! index of top cloud level for current column
-   integer,  intent(in)    :: lun    ! logical unit for diagnostic output
-   integer,  intent(in)    :: lchnk  ! chunk index
 
    logical,  intent(in)    :: doconvproc_extd(pcnst_extd)  ! indicates which species to process
-   integer,  intent(in)    :: species_class(:) ! REASTER 08/05/2015
+   integer,  intent(in)    :: species_class(:) 
```

Comment:
Is it worth adding a comment describing what a "species class" is?

Reply:
Yes, I think it will be helpful. @tangshuaiqi: I think you probably know but species class tells us what kind of species it is, e.g., aerosols or gas or cldphysics, etc. They are defined in components/eam/src/control/physconst.F90 and initialized in modal_aero_inititalized_data.F90.